kie-pmml-trusty/kie-pmml-evaluator/kie-pmml-evaluator-core/src/main/java/org/kie/pmml/evaluator/core/utils/PostProcess.java

```
@@ -123,90 +121,41 @@ static void populateOutputFields(final PMML4Result toUpdate,
-        }
-        Optional<Object> variableValue = Optional.ofNullable(outputField.evaluateTransformedValue(processingDTO));
-        variableValue.ifPresent(objValue -> addVariable(toUpdate, processingDTO, outputField.getName(), objValue));
-    }
+            case TRANSFORMED_VALUE:
+                return outputField.evaluateTransformedValue(processingDTO);
 
-    static void populateReasonCodeOutputField(final KiePMMLOutputField outputField,
-                                              final PMML4Result toUpdate,
-                                              final ProcessingDTO processingDTO) {
-        logger.debug("populateReasonCodeOutputField {} {} {}", outputField, toUpdate, processingDTO);
-        if (!RESULT_FEATURE.REASON_CODE.equals(outputField.getResultFeature())) {
-            throw new KiePMMLException("Unexpected " + outputField.getResultFeature());
-        }
-        Optional<Object> variableValue = Optional.ofNullable(outputField.evaluateReasonCodeValue(processingDTO));
-        variableValue.ifPresent(objValue -> addVariable(toUpdate, processingDTO, outputField.getName(), objValue));
-    }
+            case REASON_CODE:
+                return outputField.evaluateReasonCodeValue(processingDTO);
 
-    private static void addVariable(PMML4Result toUpdate, ProcessingDTO processingDTO, String variableName, Object variableValue) {
-        toUpdate.addResultVariable(variableName, variableValue);
-        processingDTO.addKiePMMLNameValue(new KiePMMLNameValue(variableName, variableValue));
-    }
+            case PREDICTED_DISPLAY_VALUE:
+                return model.getPredictedDisplayValue();
+
+            case ENTITY_ID:
+            case CLUSTER_ID:
+                return model.getEntityId();
+
+            case AFFINITY:
+            case ENTITY_AFFINITY:
+            case CLUSTER_AFFINITY:
+                return model.getAffinity();
 
+            default:
+                return null;
```

Shouldn't we rather throw an exception or at least log some warning?

---

I had the same idea but wanted to discuss with you. Since I got twice the same comment I implemented it now. :+1: